doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,436 @@
+         Firmware signature verified.
+         Firmware version 1
+         Setting monotonic counter (version: 1, slot: 0)
+
+   .. tab:: MCUboot
+
+      .. code-block::
+
+         [00:00:00.359,039] <inf> mcuboot: Starting bootloader
+         [00:00:00.365,295] <inf> mcuboot: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
+         [00:00:00.375,671] <inf> mcuboot: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
+         [00:00:00.386,169] <inf> mcuboot: Boot source: none
+         [00:00:00.391,815] <inf> mcuboot: Swap type: none
+         [00:00:00.420,166] <inf> mcuboot: Bootloader chainload address offset: 0xc000
+         [00:00:00.428,039] <inf> mcuboot: Jumping to the first image slot
+
+When compiled with minimal configurations that disable logging output, such as ``prj_minimal.conf``, the bootloader information output can be disabled altogether or per-bootloader.
+Refer to the source code directories of each bootloader to see what minimal configuration options are already available, or add them through a custom Kconfig fragment if desired.
+
+.. _ug_bootloader_adding_fragments:
+
+Bootloader configurations and fragments
+***************************************
+
+You can use overlays specific to bootloaders in combination with project configurations and Kconfig fragments using :ref:`ug_multi_image_variables`.
+
+For example, you can assign the ``my-custom-overlay.conf`` fragment to the |NSIB|, which uses the child name of ``b0``, as follows:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf5840 zephyr/samples/hello_world -- \
+   -DCONFIG_SECURE_BOOT=y \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   -Db0_OVERLAY_CONFIG=my-custom-overlay.conf
+
+In the same way, you can replace ``b0`` with ``mcuboot`` to apply the ``my-custom-overlay.conf`` fragment to the MCUboot image.
+This is applicable for any child image in the build, as well as any Kconfig option given from the command line.
+
+See :ref:`ug_multi_image_variables` for more information about customizing images further with this method.
+
+.. _ug_bootloader_adding_external_flash:
+
+External flash memory partitions
+********************************
+
+It is possible to use the external flash memory as the storage partition for the secondary slot.
+This requires a driver for the external flash memory that supports the following:
+
+* Single-byte read and write.
+* Writing data from the internal flash memory to the external flash memory.
+
+See :ref:`pm_external_flash` for general information about how to set up partitions in the external flash memory in the Partition Manager.
+
+.. note::
+
+   External flash memory partitions for image slots are currently only supported by MCUboot.
+
+MCUboot
+=======
+
+To configure MCUboot to use the external flash memory for the secondary slot, update the :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml` file to contain the following definition for ``mcuboot_secondary``:
```

Comment:
Maybe we could be more generic here and inform about editing configuration file for the mcuboot child image in general (adding `ncs/bootloader/mcuboot/boot/zephyr/pm.yml` as an example of such file)?

Reply:
@kyberdin

Reply:
@MarekPieta Generalized the text and used MCUboot as an example. Please see the latest commit.